Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/exchange/transform/uddi/UDDITransform.java

```diff
@@ -0,0 +1,285 @@
+                if ((UDD_SERVICE_NAMES_KEY).equals(reference.getTModelKey())) {
+                    serviceNames.add(reference.getKeyValue());
+                }
+            }
+            endpoint.getName().addAll(serviceNames);
+        }
+        return serviceNames;
+    }
+
+    private List<EndpointConfigurationType> buildEndpointConfiguration(BindingTemplate bTemplate) {
+        List<EndpointConfigurationType> epConfigList = null;
+        List<String> serviceVersions = getServiceVersion(bTemplate);
+        if (CollectionUtils.isNotEmpty(serviceVersions)) {
+            epConfigList = new ArrayList<>();
+            for (String version : serviceVersions) {
+                EndpointConfigurationType epConfig = new EndpointConfigurationType();
+                epConfig.setVersion(version);
+                epConfig.setUrl(bTemplate.getAccessPoint() != null ? bTemplate.getAccessPoint().getValue() : null);
+                epConfigList.add(epConfig);
+            }
+        }
+        return epConfigList;
+    }
+
+    private static List<String> getServiceVersion(BindingTemplate bTemplate) {
+        List<String> versions = null;
+        if (bTemplate.getCategoryBag() != null && CollectionUtils.isNotEmpty(bTemplate.getCategoryBag().
+            getKeyedReference())) {
+            versions = new ArrayList<>();
+            for (KeyedReference keyRef : bTemplate.getCategoryBag().getKeyedReference()) {
+                if (UDDI_SPEC_VERSION_KEY.equalsIgnoreCase(keyRef.getTModelKey())) {
+                    versions.add(keyRef.getKeyValue());
+                }
+            }
+        }
+        return versions;
+    }
+
+    private void buildContacts(BusinessEntity entity, OrganizationType org) {
+        List<ContactType> contacts;
+        if (entity.getContacts() != null && CollectionUtils.isNotEmpty(entity.getContacts().getContact())) {
+            contacts = new ArrayList<>();
+            for (Contact contact : entity.getContacts().getContact()) {
+                ContactType cType = new ContactType();
+                cType.getFullName().addAll(buildPersonNames(contact));
+                cType.getEmail().addAll(buildEmails(contact));
+                cType.getPhone().addAll(buildPhones(contact));
+                cType.getAddress().addAll(buildAddresses(contact));
+                cType.setRole(buildRole(contact));
+                contacts.add(cType);
+            }
+            org.getContact().addAll(contacts);
+        }
+    }
+
+    private static void buildTargetRegions(BusinessEntity entity, OrganizationType org) {
+        List<String> regions;
+        if (entity.getCategoryBag() != null && CollectionUtils.isNotEmpty(entity.getCategoryBag().getKeyedReference())) {
+            regions = new ArrayList<>();
+            for (KeyedReference keyRef : entity.getCategoryBag().getKeyedReference()) {
+                if (null != keyRef.getTModelKey() && keyRef.getTModelKey().equalsIgnoreCase(UDDI_STATE_KEY)) {
```

Comment:
remove null check here and use reverse check "constant.equalsignorecases(variable)"